src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/TransactionForRpc.cs

```diff
@@ -76,55 +81,45 @@ public JsonConverter()
             RegisterTransactionType<SetCodeTransactionForRpc>();
         }
 
-        internal static void RegisterTransactionType<T>() where T : TransactionForRpc, ITxTyped
-            => _types[(int)T.TxType] = typeof(T);
+        internal static void RegisterTransactionType<T>() where T : TransactionForRpc, IFromTransaction<T>, ITxTyped
+        {
+            Type txType = typeof(T);
+            _fromTransactionFuncs[(byte)T.TxType] = T.FromTransaction;
+            string[] uniqueProperties = txType.GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly).Where(p => p.GetCustomAttribute<JsonDiscriminatorAttribute>() is not null).Select(p => p.Name).ToArray();
```

Comment:
I am wondering if we can make it simpler, it's hard for me to understand how the the type is determined now based on fields. Let's discuss